### PR TITLE
fix ignore_file_pattern

### PR DESCRIPTION
**Pull request by zhyncs**
fix https://github.com/modelscope/modelscope/pull/866

v1.14.0 is ok
v1.15.0 `ignore_file_pattern` not effect

@wenmengzhou @liuyhwangyh May you help review this pr? Thanks.

### CODE REVIEW

**Comment by zhyncs** [comment]
cc @yingdachen

**Comment by zhyncs** [comment]
ref https://github.com/vllm-project/vllm/pull/5588/files#diff-80b1b037d9c7cb0506fdcb6a8ca72b0fb26d88fe2e3f62a058d1c45fd7cc5143R403

**Comment by wenmengzhou** [comment]
thanks for you contribution, We will review this PR this week asap

**Comment by wenmengzhou** [comment]
hi  @zhyncs  can you provide a test case for this fix?

**Comment by zhyncs** [comment]
> hi  @zhyncs  can you provide a test case for this fix?

ok

**Comment by zhyncs** [comment]
> hi @zhyncs can you provide a test case for this fix?

@wenmengzhou done

**Comment by zhyncs** [comment]
Hi @wenmengzhou Both lint and unit test have passed. May you help review the pr? Thanks.

**Comment by zhyncs** [comment]
> Hi @wenmengzhou Both lint and unit test have passed. May you help review the pr? Thanks.

cc @liuyhwangyh

File: modelscope/hub/snapshot_download.py [hunk 1]
```diff
@@ -124,7 +125,7 @@ def snapshot_download(
 
         for model_file in model_files:
             if model_file['Type'] == 'tree' or \
-                    any(fnmatch.fnmatch(model_file['Path'], pattern) for pattern in ignore_file_pattern):
+                    any([re.search(pattern, model_file['Name']) is not None for pattern in ignore_file_pattern]):
```
[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py:128]: > 如果都兼容把or any(fnmatch.fnmatch(model_file['Path'], pattern) for pattern in ignore_file_pattern)也放在条件里

Do you mean to check both Name and Path?

[the remark this answers is not in the document]
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:128]: yes, compatible with re爱你的file name

[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py]: @liuyhwangyh May you help trigger the workflows? Thanks.

[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py:128]: @liuyhwangyh 

```python3
import fnmatch
print(fnmatch.fnmatch("model.safetensors", "*.safetensors"))
# True
print(fnmatch.fnmatch("model.safetensors", "*.*.safetensors"))
# False
```

In order to be compatible with the pattern `*.*.safetensors`, it is necessary to use `re`. Currently, it is supported in hf and modelscope 1.14, but switching to `fnmatch` will cause incompatibility. So I revert the commit. cc @wenmengzhou

[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py:128]: `snapshot_download` is not on the critical path, and the performance of `re` is also acceptable.

[the remark this answers is not in the document]
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:128]: In this way, both sides are compatible:
    
    for model_file in model_files:
            if model_file['Type'] == 'tree' or \
                    any(fnmatch.fnmatch(model_file['Path'], pattern) for pattern in ignore_file_pattern) or \
                    any([re.search(pattern, model_file['Name']) is not None for pattern in ignore_file_pattern]):

[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py:128]: It looks like `model_file['Path']` and `model_file['Name']` are the same. Should we check both?

[the remark this answers is not in the document]
Comment (liuyhwangyh) [reply modelscope/hub/snapshot_download.py:128]: The two are different. model_file['Path'] contains the full path relative to the root directory， model_file['Name'] only file name.

[the remark this answers is not in the document]
Comment (zhyncs) [reply modelscope/hub/snapshot_download.py:129]: @liuyhwangyh updated

**Review by liuyhwangyh — APPROVED** [review]
LGTM

**Comment by zhyncs** [comment]
Sorry I didn't notice that the user wasn't set when I [reverted the commit](https://github.com/modelscope/modelscope/pull/886/commits/a48bd6bb2093bd7dfc5fbfd40ef7edd8dcfa1af6) yesterday, which caused it to display
`Co-authored-by: root <root@kunkun>`. 
kunkun is my girlfriend's name. My apologies for the inconvenience.

**Pull request merged**